src/main/java/com/base/site/controllers/DailyLogController.java

```diff
@@ -62,43 +82,73 @@ public String dailyLog(DailyLog dailyLog, Model model,@Param("keyword") String k
         log.info("Getmapping called for dailylog for specific date: "+date);
         Authentication auth = SecurityContextHolder.getContext().getAuthentication();
         Users loggedInUser = usersService.findByUserName(auth.getName());
 
-        List<DailyLog> logList = dailyLogService.findAll();
-        ArrayList<DailyLog> logListUserDate = new ArrayList<DailyLog>();
+        List<DailyLog> dailyLogs = dailyLogService.findAll();
+        List<DailyLog> dailyLogsView = new ArrayList<>();
+        //List<Exercise> exerciseList = exerciseService.findAllByKeyword(keyword);
 
-        LocalDate enteredDate = LocalDate.parse(date);
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
+        Date enteredDate = dateFormat.parse(date);
 
-        for (DailyLog logdate: logList) {
-            if(logdate.getDatetime().equals(enteredDate) && loggedInUser.getId() == logdate.getFkUser().getId() && logdate.getFood() != null) {
-                logListUserDate.add(logdate);
+        for (DailyLog dailyLog: dailyLogs) {
+            String timeStamp = dailyLog.getDatetime().format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+            if(timeStamp.equals(date) && loggedInUser.getId() == dailyLog.getFkUser().getId()) {
+                String sDatetime = dailyLog.getDatetime().format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+                dailyLog.setSDatetime(sDatetime);
+                dailyLogsView.add(dailyLog);
             }
         }
 
-        model.addAttribute("list",logListUserDate);
+        LocalDate localDate = enteredDate.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+
+        Date currentDate = new Date();
+        LocalDate today = currentDate.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+        model.addAttribute("today", today.format(DateTimeFormatter.ofPattern("dd-MM-yyyy")));
+        model.addAttribute("sSelectedDate", localDate.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")));
+
+        model.addAttribute("list", dailyLogsView);
         model.addAttribute("keyword", keyword);
 
+
+        // +/- Day
+        model.addAttribute("tomorrow", localDate.plusDays(1).format(DateTimeFormatter.ofPattern("dd-MM-yyyy")));
+        model.addAttribute("yesterday", localDate.minusDays(1).format(DateTimeFormatter.ofPattern("dd-MM-yyyy")));
+
+        // +/- Week
+        model.addAttribute("nextWeek", localDate.plusWeeks(1).format(DateTimeFormatter.ofPattern("dd-MM-yyyy")));
+        model.addAttribute("previousWeek", localDate.minusWeeks(1).format(DateTimeFormatter.ofPattern("dd-MM-yyyy")));
+
+        // +/- Month
+        model.addAttribute("nextMonth", localDate.plusMonths(1).format(DateTimeFormatter.ofPattern("dd-MM-yyyy")));
+        model.addAttribute("previousMonth", localDate.minusMonths(1).format(DateTimeFormatter.ofPattern("dd-MM-yyyy")));
+
+        //model.addAttribute("bmr", loggedInUser.getBMR(usersService.getLatestWeight(localDate)));
```

Comment:
I had to out comment this because it expected a double but got a DailyLog instead.

Reply:
we can just add .getDate so it looks like this loggedInUser.getBMR(usersService.getLatestWeight(localDate)).getDate() and it should work